README.md

```diff
@@ -7,59 +7,661 @@
+on a system that doesn't support conditional transfers. Hashed timelock transfers
+go through plugin virtual like a clearing layer, and every so often a settlement
+is sent to make sure the amount secured on the ledger doesn't get too far from
+the finalized amount owed.
 
-If you want to generate a trustline prefix from a secret key, peer's public key, and currency,
-you can access the method through the token class:
+Unlike creating a claim, sending a payment has side-effects (it alters an
+external system). Therefore, the code is slightly more complicated.
 
 ```js
-const Token = require('ilp-plugin-virtual/src/util/token')
-const prefix = Token.prefix({
-  secretKey: 'o9Lt0oZFek1ArM_A0HUAq8M8edRuoSeLjX8i10sVUiY',
-  peerPublicKey: 'KRixgcBCBdyQln7IBYiopjuNO78QSFtXgOwP1sbsCSk',
-  currency: 'USD'
-})
+const { MakePaymentChannelPlugin } = require('ilp-plugin-virtual')
+const { NotAcceptedError } = require('ilp-plugin-shared').Errors
+const Network = require('some-example-network')
+const BigNumber = require('bignumber.js')
+
+return MakePaymentChannelPlugin({
+  // initialize fields and validate options in the constructor
+  constructor: function (ctx, opts) {
+    // In this type of payment channel module, we create a log of incoming
+    // settlements to track all the transfers sent to us on the ledger we're
+    // using for settlement.  We use a transferLog in order to make sure a
+    // single transfer can't be added twice.
+    ctx.state.incomingSettlements = ctx.backend.getTransferLog('incoming_settlements')
+
+    // The amount settled is used to track how much we've paid out in total.
+    // We'll go deeper into how it's used in the `createOutgoingClaim`
+    // function.
+    ctx.state.amountSettled = ctx.backend.getMaxValueTracker('amount_settled')
+
+    // In this type of payment channel backend, the unsecured balance we want
+    // to limit is the total amount of incoming transfers minus the sum of all
+    // the settlement transfers we've received.
+    ctx.state.maxUnsecured = opts.maxUnsecured
+
+    // use some preconfigured secret for authentication
+    ctx.state.authToken = opts.authToken
+  },
+
+  getAuthToken: (ctx) => (ctx.state.authToken),
+
+  connect: async function (ctx, opts) {
+    await Network.connectToNetwork()
+
+    // establish metadata during the connection phase
+    ctx.state.prefix = 'peer.network.' + (await Network.getChannelId())
+    ctx.state.account = await Network.getChannelSource()
+    ctx.state.peer = await Network.getChannelDestination()
+    ctx.state.info = {
+      prefix: ctx.state.prefix,
+      currencyScale: 6,
+      currencyCode: 'X??',
+      connectors: []
+    }
+  },
+
+  // Synchronous functions in order to get metadata. They won't be called until
+  // after the plugin is connected.
+  getAccount: (ctx) => (ctx.state.prefix + ctx.state.account),
+  getPeerAccount: (ctx) => (ctx.state.prefix + ctx.state.peer),
+  getInfo: (ctx) => (ctx.state.info),
```

Comment:
added code to auto-clone the result of getInfo in the framework code, and added a note in the method description that the result will be cloned